src/pybind/mgr/dashboard_v2/services/ceph_service.py

```diff
@@ -1,32 +1,112 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
 
-from . import Service
+import json
+import rbd
 
+from ..tools import ViewCache
+from .. import logger
+
+
+def _get_service_map(mgr, service_name):
+    service_map = {}
+    for server in mgr.list_servers():
+        for service in server['services']:
+            if service['type'] == service_name:
+                if server['hostname'] not in service_map:
+                    service_map[server['hostname']] = {
+                        'server': server,
+                        'services': []
+                    }
+                inst_id = service['id']
+                metadata = mgr.get_metadata(service_name, inst_id)
+                status = mgr.get_daemon_status(service_name, inst_id)
+                service_map[server['hostname']]['services'].append({
+                    'id': inst_id,
+                    'type': service_name,
+                    'hostname': server['hostname'],
+                    'metadata': metadata,
+                    'status': status
+                })
+    return service_map
+
+
+def _get_pool_list(mgr, application=None):
+    osd_map = mgr.get('osd_map')
+    if not application:
+        return osd_map['pools']
+    return [pool for pool in osd_map['pools']
+            if application in pool.get('application_metadata', {})]
+
+
+def _get_daemons(mgr):
+    daemons = []
+    for hostname, server in _get_service_map(mgr, 'rbd-mirror').items():
+        for service in server['services']:
+            id = service['id']  # pylint: disable=W0622
+            metadata = service['metadata']
+            status = service['status']
+
+            try:
+                status = json.loads(status['json'])
+            except ValueError:
+                status = {}
+
+            instance_id = metadata['instance_id']
+            if id == instance_id:
+                # new version that supports per-cluster leader elections
+                id = metadata['id']
+
+            # extract per-daemon service data and health
+            daemon = {
+                'id': id,
+                'instance_id': instance_id,
+                'version': metadata['ceph_version'],
+                'server_hostname': hostname,
+                'service': service,
+                'server': server,
+                'metadata': metadata,
+                'status': status
+            }
+            daemon = dict(daemon, **_get_daemon_health(daemon))
+            daemons.append(daemon)
+
+    return sorted(daemons, key=lambda k: k['instance_id'])
+
+
+def _get_daemon_health(daemon):
```

Comment:
The name is misleading: `_get_daemon_health` doesn't return anything that is related to "health" of a generic daemon. Instead, it is tightly bound to the rbd-mirroring controller.